IOMC/RandomEngine/src/RandomNumberGeneratorService.cc

```diff
@@ -150,20 +151,36 @@ namespace edm {
         else {
           if(initialSeedSet.size() != 1U) {
             throw Exception(errors::Configuration)
-              << "Random engines of type \"HepJamesRandom\" and \"TRandom3\n"
+              << "Random engines of type \"HepJamesRandom\", \"TRandom3\" and \"MixMaxRng\" \n"
               << "require exactly 1 seed be specified in the configuration.\n"
               << "There were " << initialSeedSet.size() << " seeds set for the\n"
               << "module with label \"" << label << "\".\n" ;
           }
           if(engineName == "HepJamesRandom") {
             if(initialSeedSet[0] > maxSeedHepJames) {
               throw Exception(errors::Configuration)
-                << "The CLHEP::HepJamesRandom engine seed should be in the range 0 to 900000000.\n"
+                << "The CLHEP::HepJamesRandom engine seed should be in the range 0 to " << maxSeedHepJames <<".\n"
                 << "The seed passed to the RandomNumberGenerationService from the\n"
                    "configuration file was " << initialSeedSet[0] << ".  This was for \n"
                 << "the module with label " << label << ".\n";
             }
-          } else if(engineName != "TRandom3") {
+	  } else if(engineName == "MixMaxRng") {
+            if(initialSeedSet[0] > maxSeedTRandom3) {
+              throw Exception(errors::Configuration)
+                << "The CLHEP::MixMaxRng engine seed should be in the range 0 to " << maxSeedTRandom3 << ".\n"
+                << "The seed passed to the RandomNumberGenerationService from the\n"
+                   "configuration file was " << initialSeedSet[0] << ".  This was for \n"
+                << "the module with label " << label << ".\n";
+            }
+	  } else if(engineName == "TRandom3") {
+            if(initialSeedSet[0] > maxSeedTRandom3) {
+              throw Exception(errors::Configuration)
+                << "The CLHEP::MixMaxRng engine seed should be in the range 0 to " << maxSeedTRandom3 << ".\n"
+                << "The seed passed to the RandomNumberGenerationService from the\n"
+                   "configuration file was " << initialSeedSet[0] << ".  This was for \n"
+                << "the module with label " << label << ".\n";
+            }
+          } else {
```

Comment:
I reviewed the service code and the test module code. They look good. The only minor things I noticed were that in this section the indentation seems off by a space (or maybe it is just my browser or could be caused by a tab that needs replacing). It also seems to me to be unnecessary to check that the seeds are not too large for the new engine and TRandom3, because they are stored in variables that are explicitly 32 bits and the largest value is the maximum 32 bits can hold ...  I didn't look at the unit test configuration files yet because the unit tests are still failing, but my first guess as to the problem would be that I do not see that the reference files in the unit_test_outputs subdirectory were updated. It has been so long since I looked at these I do not remember this clearly, but probably you need to run the tests with "scram b runtests" and manually check the differences in the output text files and then copy the output to overwrite the reference files.